tools/d2g/d2gtest/testdata/testcases/env_escaping_test.yml

```diff
@@ -0,0 +1,38 @@
+---
+testSuite:
+  name: Environment variable escaping test
+  description: Test that environment variables are properly escaped.
+  payloadTests:
+    - name: Env escaping test
+      description: >-
+        If an environment variable contains spaces or special characters,
+        it should be escaped before passing to the podman run command.
+      dockerWorkerTaskPayload:
+        command:
+          - echo "Hello world"
+        image: ubuntu
+        maxRunTime: 3600
+        env:
+          test123 --help ; whoami ;: testing
+      genericWorkerTaskPayload:
+        command:
+          - - bash
+            - '-cx'
+            - >-
+              podman run -t --rm
+              -e 'test123 --help ; whoami ;'
+              -e RUN_ID
+              -e TASKCLUSTER_ROOT_URL
+              -e TASKCLUSTER_WORKER_LOCATION
+              -e TASK_ID
+              ubuntu 'echo "Hello world"'
+        env:
+          test123 --help ; whoami ;: testing
```

Comment:
Is this correct? I'm surprised there is no string quoting here too. When this is converted to json, what does it look like? My understanding of yaml isn't sufficient to know how this is parsed.

Reply:
Yeah, json output of this is:

```json
"env": {
    "test123 --help ; whoami ;": "testing"
},
```

Reply:
Looks like it strips trailing spaces

Reply:
Added in https://github.com/taskcluster/taskcluster/pull/6660/commits/f4ae953781fcc8d286f2748a069183afbcc2fea4.